lib/utils/config/definitions.js

```diff
@@ -1239,7 +1239,7 @@ define('location', {
 
 define('lockfile-version', {
   default: null,
-  type: [null, 1, 2, 3, '1', '2', '3'],
+  type: [null, 1, 2, 3],
```

Comment:
they aren’t duplicates, they’re strings vs numbers.

Reply:
Yeah without making a custom data type this has to stay like it is.  We have to support both npmrc file entries which are typed, and cli input which is only ever strings.

Reply:
The log warning would be where this could be cleaned up, if at all.

Reply:
@wraithgar @ljharb 
Thank you for reviewing and feedback. Apologies for misunderstanding the use of `type` here. Now I'm clear on that. However, where should I look for the log warnings? As I'm new to this repo, could you please guide me to look into the correct path?

Reply:
Sure thing. It's https://github.com/npm/cli/blob/baae2d60109c3362c38ad09c231a32c84c26f63a/workspaces/config/lib/index.js#L535-L538

Bonus points for every ternary you remove from `invalidHandler` in general.

Reply:
Hi @wraithgar, have updated the log warning as suggested and reverted back that definition type changes. Also have refactored some of the ternary code. Please review and let me know your comments. Thank you!